### PR TITLE
Improve Performance: replace bitmap.colorAt with js implemention

### PR DESCRIPTION
**Pull request by Nihiue**
Bitmap.ColorAt is extremely slow , due to the  overhead of calling native addons.

So I replaced it with pure  js implemention, which is 10x-50x faster

Original Issue
https://github.com/octalmage/robotjs/issues/648

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,9 +13,28 @@ function bitmap(width, height, byteWidth, bitsPerPixel, bytesPerPixel, image)
...
+        return n.toString(16).padStart(2, '0');
+    }
+
+
     this.colorAt = function(x, y)
     {
-        return robotjs.getColor(this, x, y);
+        const buffer = this.image;
+        const startIndex = (y * this.width + x) * this.bytesPerPixel;
+        if (typeof x !== 'number' || typeof y !== 'number') {
```
Comment (saghul) [inline index.js:26]: Maybe check this earlier, before using it to calculate startIdnex?

File: index.js [hunk 2]
```diff
@@ -13,9 +13,28 @@ function bitmap(width, height, byteWidth, bitsPerPixel, bytesPerPixel, image)
     this.bytesPerPixel = bytesPerPixel;
     this.image = image;
 
+    function toHex(n)
```
Comment (saghul) [inline index.js:16]: doesn't it make sense for this function to be out of here?

File: index.js [hunk 3]
```diff
@@ -13,9 +13,28 @@ function bitmap(width, height, byteWidth, bitsPerPixel, bytesPerPixel, image)
...
-        return robotjs.getColor(this, x, y);
+        const buffer = this.image;
+        const startIndex = (y * this.width + x) * this.bytesPerPixel;
+        if (typeof x !== 'number' || typeof y !== 'number') {
+            throw new Error(`Invalid number`);
+        }
+        if (x < 0 || x >= this.width || y < 0 || y >= this.height || typeof buffer[startIndex + 2] === 'undefined') {
+            throw new Error(`(${x}, ${y}) are outside the bitmap`);
+        }
+        let ret = '';
```
Comment (saghul) [inline index.js:32]: Is this a specific optimization or could we use a string template instead?

**Review by saghul — COMMENTED** [review]
Left a few comments.

**Comment by Nihiue** [comment]
Refined code style according to comments.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Thanks for the reminder!

**Pull request merged**